signac/core/mongodbcollection.py

```diff
@@ -0,0 +1,147 @@
+        import pymongo
+
+        if client is None:
+            mongo_kwargs = mongo_kwargs if mongo_kwargs is not None else {}
+            self._client = pymongo.MongoClient(**mongo_kwargs)
+        else:
+            self._client = client
+        self._db = self._client.get_database(database)
+        self._collection = self._db.get_collection(collection)
+        self._name = name
+        self._key = type(self).__name__ + '::name'
+        super().__init__(**kwargs)
+        if (name is None) == (self._parent is None):
+            raise ValueError(
+                "Illegal argument combination, one of the two arguments, "
+                "parent or name must be None, but not both.")
+
+    def _load(self):
+        """Load the data from a Mongo-database."""
+        blob = self._collection.find_one({self._key: self._name})
+        return blob['data'] if blob is not None else None
+
+    def _sync(self):
+        """Write the data from Mongo-database."""
+        data = self.to_base()
+        data_to_insert = {self._key: self._name, 'data': data}
+        self._collection.replace_one({self._key: self._name}, data_to_insert, True)
+
+
+class MongoDict(MongoCollection, SyncedAttrDict):
+    """A dict-like mapping interface to a persistent Mongo-database.
+
+    The MongoDict inherits from :class:`~core.collection_api.MongoCollection`
+    and :class:`~core.syncedattrdict.SyncedAttrDict`.
+
+    .. code-block:: python
+
+        doc = MongoDict('data')
+        doc['foo'] = "bar"
+        assert doc.foo == doc['foo'] == "bar"
+        assert 'foo' in doc
+        del doc['foo']
+
+    .. code-block:: python
+
+        >>> doc['foo'] = dict(bar=True)
+        >>> doc
+        {'foo': {'bar': True}}
+        >>> doc.foo.bar = False
+        {'foo': {'bar': False}}
+
+    .. warning::
+
+        While the MongoDict object behaves like a dictionary, there are
+        important distinctions to remember. In particular, because operations
+        are reflected as changes to an underlying database, copying (even deep
+        copying) a MongoDict instance may exhibit unexpected behavior. If a
+        true copy is required, you should use the `to_base()` method to get a
+        dictionary representation, and if necessary construct a new MongoDict
+        instance: `new_dict = MongoDict(old_dict.to_base())`.
+
+    Parameters
+    ----------
+    name: str
+        The name of the  collection (Default value = None).
+    client: object
+        A Mongo client.
+    mongo_kwargs: dict
+        kwargs arguments passed through to the `pymongo.MongoClient` function.
+    database : string
+        Name of database (Default value = 'signac_db').
+    collection : string
+        Name of collection (Default value = 'collection')
+    data: mapping, optional
+        The intial data pass to MongoDict. Defaults to `dict()`
+    parent: object, optional
+        A parent instance of MongoDict or None (Default value = None).
+    """
+    pass
```

Comment:
Looks like A+ abstraction here. 👍 
Do we even need the `pass` statement here?

Reply:
No, we don't need that. I have removed it.